### PR TITLE
Add merge points

### PR DESCRIPTION
**Pull request by akaszynski**
Address #4345 by adding `merge_points`.

```py
>>> import pyvista as pv
>>> cyl = pv.CylinderStructured(radius=[0, 0.1, 0.2])
>>> cyl.n_points
990
>>> cyl_merged = cyl.merge_points()
>>> cyl_merged.n_points
670
```

### CODE REVIEW

File: pyvista/core/filters/data_set.py [hunk 1]
```diff
@@ -4722,6 +4722,43 @@ def merge(
...
+        >>> import pyvista as pv
+        >>> cyl = pv.CylinderStructured(radius=[0, 0.1, 0.2])
+        >>> cyl.n_points
+        990
+        >>> cyl_merged = cyl.merge_points()
+        >>> cyl_merged.n_points
+        670
+
+        """
+        return self.merge(tolerance=tolerance, progress_bar=progress_bar)
```
Comment (tkoyama010) [inline pyvista/core/filters/data_set.py:4760]: I agree that this name is better, but what about deprecating the `merge` method and replacing it with `merge_points`?
In DataSet, I think only `merge_points` is sufficient.

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

File: pyvista/core/filters/data_set.py (hunk 1 above)
Comment (akaszynski) [reply pyvista/core/filters/data_set.py:4760]: `merge` is still necessary in order to merge multiple datasets. `merge_points` is just for merging points of the dataset itself.

**Review by tkoyama010 — APPROVED** [review]
LGTM

File: pyvista/core/filters/data_set.py (hunk 1 above)
Comment (tkoyama010) [reply pyvista/core/filters/data_set.py:4760]: Got it. Thanks!

**Review by banesullivan — CHANGES_REQUESTED** [review]
I think this should be called `clean()`. It is effectively the  `vtkCleanPolyData` filter but for non-poly data. We have `clean()` as a method specific to `PolyData`, and I think it would be better to have this method be the default `clean()` method for all other data types consider this is missing

**Comment by akaszynski** [comment]
I just learned that there's [vtkStaticCleanUnstructuredGrid](https://vtk.org/doc/nightly/html/classvtkStaticCleanUnstructuredGrid.html), which might be what we want to use instead as a `clean` method as proposed by @banesullivan. However, I'm wary of implementing this as a `clean` method at the `DataSet` level since this will violate the [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle).

**Comment by banesullivan** [comment]
Fair enough!

This meme explains [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle): 
ilxzO

**Comment by akaszynski** [comment]
Dropping in favor of #4367.

**Pull request closed without merging**